docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,771 @@
+
+image::auth0-created-application.png[Created Auth0 application]
+
+Next, while still in the `Auth0` dashboard, add some users to your application.
+
+image::auth0-add-user.png[Add Auth0 application users]
+
+Now that you have successfully created and configured your `Auth0` application, you are ready to start creating and configuring a Quarkus endpoint. 
+In the steps that follow, you will continue to configure and update your `Auth0` application as needed.
+
+== Create a Quarkus application
+
+Use the following Maven command to create a Quarkus RESTEasy Reactive application that can be secured with the Quarkus OIDC extension.
+
+:create-app-artifact-id: quarkus-auth0
+:create-app-extensions: resteasy-reactive,oidc
+include::{includes}/devtools/create-app.adoc[]
+
+Create the application workspace and import it into your favorite IDE.
+A `GreetingResource` JAX-RS resource gets generated, which you will need to secure.
+
+[source,java]
+----
+package org.acme;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello from RestEasy Reactive";
+    }
+}
+----
+
+Update the Quarkus application code so that only authenticated users can access the `hello()` method and update it to return the current username instead, as shown in the following example:
+
+[source,java]
+----
+package org.acme;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.oidc.IdToken;
+import io.quarkus.security.Authenticated;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @Inject
+    @IdToken                                        <1>
+    JsonWebToken idToken;
+	
+    @GET
+    @Authenticated                                  <2>
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello, " + idToken.getName();
+    }
+}
+----
+<1> The injected `JsonWebToken` (JWT) token has an `@IdToken` qualifier.
+`IdToken` represents the user authentication part of the OIDC authorization code flow mechanism.
+<2> The `io.quarkus.security.Authenticated` annotation is added to the `hello()` method, which means that only authenticated users can access it.
+
+[NOTE]
+====
+The access token acquired during the authorization code flow, alongside the ID token, is not used directly by the endpoint but is only used to access downstream services on behalf of the currently authenticated user. 
+More to come on the topic of "access tokens", later in this tutorial.
+====
+
+Configure OIDC in `application.properties` with the properties from the created `Auth0` application:
```

Comment:
As there are two platforms being used in this tutorial, would it help to set the context in places, for example:

```suggestion
Configure OIDC in the Quarkus `application.properties` file  by using the properties from the Auth0 application that you created earlier.
```